client/asset/btc/spv.go

```diff
@@ -1244,6 +1241,57 @@ func (w *spvWallet) startWallet() error {
 	return nil
 }
 
+// deleteWalletData will delete all files except peers.json from the
+// wallet's directory.
+func (w *spvWallet) deleteWalletData() error {
```

Comment:
I have not reviewed this PR yet (very sorry), but  I would not use `os.Remove`(anything) on a file that's still opened.  Note that there's no file handle arg on that, it's just a syscall.  Before deleting anything I would close it.

Reply:
It would probably be better to use a new database folder and not delete anything, if that is doable.

Or move/rename the old folder to something like `oldName.save_date`? Then can always recover from the recover.

Here's some info: https://www.baeldung.com/linux/open-file-handle-after-move-delete

Seems to suggest, on linux, that both deleting and renaming to a file on the same disc would do what we want. The file handles will still be pointing to the old files, even if they do not exist. Do not know what happens on Windows, but I suspect deletion or renaming would fail because file is in use.

Reply:
> Here's some info: https://www.baeldung.com/linux/open-file-handle-after-move-delete

That's a good read. `spvWallet.stop()` closes all the file handles, and it is called before `Destroy`. Are you saying that in case `spvWallet.stop()` fails, I should find any file handles that were left and close them before calling `os.Remove`?

A backup folder is definitely a good idea though.

Reply:
> I should find any file handles that were left and close them before calling os.Remove?

If this was directed to me... no. It looks like from reading that that it doesn't matter if you rename or delete the file. The new folder will not be the same thing internally for linux, and Windows will probably just fail whatever you do.

The wallet could potentially not run if it is trying to listen on the same ports though, and the bad instance is still using those.